standard/class.lua

```diff
@@ -37,16 +38,23 @@ function Class.new(base, init)
 	local metatable = {}
 
 	metatable.__call = function(class_tbl, ...)
-		local varArgs = {n = select('#', ...), ...}
 		local object = {}
 		setmetatable(object, instance)
 
-		Class.construct(instance, object, varArgs)
+		instance.init(object, ...)
 
 		return object
 	end
 
-	instance.init = init
+	instance.init = function(object, ...)
+		if base and base.init then
```

Comment:
With this solution, base.init can never be nil once base exists, as every instance now has a function there.
IIRC this was a measure to catch errors where not all modules where using Class/dev yet.
```suggestion
		if base then
```